### PR TITLE
Run C#9 test cases only on .Net 5

### LINKED ISSUE #3738: Run C#9 tests only `#if NET`
There's no use case where C# 9 is going to be able to run under .NET Framework, so no need to run tests there. Or in NETCOREAPP 3.x.

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Fixes #3738

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/AvoidExcessiveClassCouplingTest.cs [hunk 1]
```diff
@@ -33,10 +33,12 @@ public class AvoidExcessiveClassCouplingTest
         public void AvoidExcessiveClassCoupling() =>
             Verifier.VerifyAnalyzer(@"TestCases\AvoidExcessiveClassCoupling.cs", new AvoidExcessiveClassCoupling { Threshold = 1 });
 
+#if NET
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/AvoidExcessiveClassCouplingTest.cs:36]: also
- AvoidExcessiveClassCoupling_InRecord_Enums_Not_Counted
- AvoidExcessiveClassCoupling_Base_Records_Interfaces_NotCounted
- AvoidExcessiveClassCoupling_Primitive_Types_Not_Counted

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DisposableMemberInNonDisposableClassTest.cs.orig [hunk 2]
```diff
@@ -0,0 +1,100 @@
+﻿/*
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DisposableMemberInNonDisposableClassTest.cs.orig:1]: this seems that it's an artefact that needs to be removed?

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]
There are also tests which use `ParseOptionsHelper.FromCSharp9` and I see they haven't been ported to the new API, like:
- CbdeHandler_CS_FromCSharp9
- CookiesShouldBeHttpOnly_Net
- CookiesShouldBeSecure_Net
- XmlExternalEntityShouldNotBeParsed_XmlReader_CSharp9

And code fixes:
- CheckFileLicenseCodeFix_CSharp9_ShouldBeNoncompliant_CS
- ConditionalSimplification_FromCSharp9_CodeFix
- EmptyStatement_CodeFix_CSharp9
- UseShortCircuitingOperators_CSharp9_CodeFix
- VirtualEventField_CSharp9_CodeFix

**Comment by costin-zaharia-sonarsource** [comment]
> There are also tests which use ParseOptionsHelper.FromCSharp9

Yes! I forgot to search for `ParseOptionsHelper.FromCSharp9` usages. 

For code fixes we don't have methods specific for C#9 and since there are just a few calls I prefer to let them this way.

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**